### DISCUSSION TITLE
Mongdb $pull operator or equivalent.

### DISCUSSION THREAD

**Discussion by arthur-alves** [prisma/prisma#16353] [Q&A] [answered]
Hi everyone,

Just to make sure, is there any way to use $pull operator from mongodb inside Prisma, even using $runCommandRaw? Unfortunately, I did not found anything related.

**Comment by nurul3101** [comment, marked as answer]
Hello @arthur-alves 👋 

Yes, you can use $pull operator of MongoDB with Prisma using `$runCommanRaw`.

For this schema:
```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider        = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model Stores {
  id         String   @id @default(auto()) @map("_id") @db.ObjectId
  name       String
  fruits     String[]
  vegetables String[]
}

```

Here's an example of using $pull operator which removes an element from vegetables array:

```typescript
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient({});

async function main() {
  await prisma.stores.deleteMany();

  await prisma.stores.create({
    data: {
      name: 'test',
      fruits: ['apple', 'banana'],
      vegetables: ['carrot', 'potato'],
    },
  });

  const commandRaw = await prisma.$runCommandRaw({
    update: 'Stores',
    updates: [
      {
        q: { name: 'test' },
        u: { $pull: { vegetables: { $in: ['carrot'] } } },
      },
    ],
  });

  console.log(JSON.stringify(commandRaw));

  const response = await prisma.stores.findMany({
    where: {
      name: 'test',
    },
  });

  console.log(JSON.stringify(response));
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

```

Response: The vegetable carrot is removed from the response
```
> ts-node index.ts

{"n":1,"nModified":1,"ok":1}
[{"id":"637c8ae1c2b64585856d69b8","name":"test","fruits":["apple","banana"],"vegetables":["potato"]}]
```

**Reply by pantharshit00** [reply]
👋 Hello @arthur-alves,  did you perhaps have a chance to check @nurul3101's recommendation? Let us know if you still have any questions!
 
 Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Reply by arthur-alves** [reply]
Yeah sorry for delay, it works great. Thanks

**Comment by arthur-alves** [comment]
Thank you @nurul3101

**Reply by nurul3101** [reply]
No worries! :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
